hooks/useLoginHook.ts

```diff
@@ -0,0 +1,158 @@
+        dispatch({
+          field: 'isPaidUser',
+          value: userProfile.plan !== 'free',
+        });
+        dispatch({ field: 'isTempUser', value: userProfile.isTempUser });
+        if (userProfile.isTempUser || userProfile.isTeacherAccount) {
+          fetchTeacherPrompts().then((res) => {
+            if (res.data) {
+              dispatch({
+                field: 'teacherPrompts',
+                value: res.data.prompts,
+              });
+            }
+          });
+        }
+        if (userProfile.isTempUser) {
+          fetchTeacherSettings().then((res) => {
+            if (res.data) {
+              dispatch({
+                field: 'teacherSettings',
+                value: res.data.settings,
+              });
+            }
+          });
+        }
+        dispatch({
+          field: 'isTeacherAccount',
+          value: userProfile.isTeacherAccount,
+        });
+        dispatch({
+          field: 'isUltraUser',
+          value: userProfile.plan === 'ultra',
+        });
+        dispatch({
+          field: 'hasMqttConnection',
+          value: userProfile.hasMqttConnection,
+        });
+        dispatch({
+          field: 'isConnectedWithLine',
+          value: userProfile.isConnectedWithLine,
+        });
+        dispatch({
+          field: 'user',
+          value: {
+            id: session.user.id,
+            email: session.user.email,
+            plan: userProfile.plan || 'free',
+            token: session.access_token,
+            referralCode: userProfile.referralCode,
+            referralCodeExpirationDate: userProfile.referralCodeExpirationDate,
+            proPlanExpirationDate: userProfile.proPlanExpirationDate,
+            hasReferrer: userProfile.hasReferrer,
+            hasReferee: userProfile.hasReferee,
+            isInReferralTrial: userProfile.isInReferralTrial,
+          },
+        });
+
+        updateUserInfo({
+          id: userProfile.id,
+          email: userProfile.email,
+          plan: userProfile.plan || 'free',
+          associatedTeacherId: userProfile.associatedTeacherId,
+          isTeacherAccount: userProfile.isTeacherAccount,
+          isTempUser: userProfile.isTempUser,
+          tempUserUniqueId: userProfile.tempUserUniqueId,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+        toast.error(
+          t('Unable to load your information, please try again later.'),
+        );
+      })
+      .finally(() => {
+        dispatch({ field: 'appInitialized', value: true });
+      });
+  };
+
+  const checkSurveyFilled = (session: Session) => {
+    supabase
+      .from('user_survey')
+      .select('name')
+      .eq('uid', session.user.id)
+      .then(({ data }) => {
+        if (!data || data.length === 0) {
+          dispatch({ field: 'isSurveyFilled', value: false });
+        } else {
+          dispatch({ field: 'isSurveyFilled', value: true });
+        }
+      });
+  };
```

Comment:
Let's remove this function, since we no longer have survey for user to fill now. Sorry didn't bring this up earlier